test/src/SpendPermissions/approveWithRevoke.t.sol

```diff
@@ -279,6 +279,81 @@ contract ApproveWithRevokeTest is SpendPermissionManagerBase {
         vm.assertTrue(mockSpendPermissionManager.isApproved(newSpendPermission));
     }
 
+    function test_approveWithRevoke_success_returnsTrueOldRevokedNewApproved(
+        address spender,
+        uint48 start,
+        uint48 end,
+        uint48 period,
+        uint160 allowance,
+        uint256 salt,
+        bytes memory extraData
+    ) public {
+        vm.assume(spender != address(0));
+        vm.assume(spender != address(account));
+        vm.assume(start > 0);
+        vm.assume(start < end);
+        vm.assume(period > 0);
+        vm.assume(allowance > 0);
+
+        SpendPermissionManager.SpendPermission memory newSpendPermission = SpendPermissionManager.SpendPermission({
+            account: address(account),
+            spender: spender,
+            token: NATIVE_TOKEN,
+            start: start,
+            end: end,
+            period: period,
+            allowance: allowance,
+            salt: salt,
+            extraData: extraData
+        });
+        vm.startPrank(address(account));
+        bool isApproved = mockSpendPermissionManager.approveWithRevoke(
+            newSpendPermission, existingSpendPermission, lastValidUpdatedPeriod
+        );
+        vm.stopPrank();
+        vm.assertTrue(isApproved);
+        vm.assertFalse(mockSpendPermissionManager.isApproved(existingSpendPermission));
+        vm.assertTrue(mockSpendPermissionManager.isApproved(newSpendPermission));
+    }
+
+    function test_approveWithRevoke_success_returnsFalseIfOldRevokedNewApprovedAfterBeingRevoked(
```

Comment:
To be extra thorough, we also want to check that if false is returned, then the permission did not have an event emitted for it.